appmail/tests/test_models.py

```diff
@@ -161,69 +161,101 @@ def test_create_message(self):
 
         message = template.create_message(
             context,
-            to=['bruce@kung.fu'],
-            cc=['fred@example.com'],
-            bcc=['ginger@example.com'],
-            from_email='Fred <fred@example.com>'
-        )
-        self.assertEqual(message.to, ['bruce@kung.fu'])
-        self.assertEqual(message.cc, ['fred@example.com'])
-        self.assertEqual(message.bcc, ['ginger@example.com'])
-        self.assertEqual(message.from_email, 'Fred <fred@example.com>')
+            to=["bruce@kung.fu"],
+            cc=["fred@example.com"],
+            bcc=["ginger@example.com"],
+            from_email="Fred <fred@example.com>",
+        )
+        self.assertEqual(message.to, ["bruce@kung.fu"])
+        self.assertEqual(message.cc, ["fred@example.com"])
+        self.assertEqual(message.bcc, ["ginger@example.com"])
+        self.assertEqual(message.from_email, "Fred <fred@example.com>")
         # and so on - not going to test every property.
 
         # but we will check the three illegal kwargs
-        self.assertRaises(AssertionError, template.create_message, {}, subject='foo')
-        self.assertRaises(AssertionError, template.create_message, {}, body='foo')
-        self.assertRaises(AssertionError, template.create_message, {}, alternatives='foo')
+        with self.assertRaisesMessage(
+            ValueError, "Invalid argument: 'subject' is set from the template."
+        ):
+            template.create_message({}, subject="foo")
+        with self.assertRaisesMessage(
+            ValueError, "Invalid argument: 'body' is set from the template."
+        ):
+            template.create_message({}, body="foo")
+        with self.assertRaisesMessage(
+            ValueError, "Invalid argument: 'alternatives' is set from the template."
+        ):
+            template.create_message({}, alternatives="foo")
+
+    def test_create_message__with_attachments__allowed(self):
+        template = EmailTemplate(
+            subject="Welcome {{ first_name }}",
+            body_text="Hello {{ first_name }}",
+            body_html="<h1>Hello {{ first_name }}</h1>",
+            supports_attachments=True,
+        )
+        template.create_message({}, attachments=[mock.Mock(spec=MIMEImage)])
+
+    def test_create_message__with_attachments__disallowed(self):
+        template = EmailTemplate(
+            subject="Welcome {{ first_name }}",
+            body_text="Hello {{ first_name }}",
+            body_html="<h1>Hello {{ first_name }}</h1>",
+            supports_attachments=False,
+        )
+        with self.assertRaisesMessage(
+            ValueError, "Email template does not support attachments."
+        ):
+            template.create_message({}, attachments=[mock.Mock(spec=MIMEImage)])
```

Comment:
This is the functional element in use - if you try to add attachments to a template that doesn't support them, it'll error.